addon/metrics-adapters/google-tag-manager.js

```diff
@@ -25,21 +25,28 @@ export default BaseAdapter.extend({
 
   init() {
     const config = get(this, 'config');
-    const { id } = config;
-    const dataLayer = getWithDefault(config,'dataLayer', 'dataLayer');
+    const { id, envParams } = config;
+    const dataLayer = getWithDefault(config, 'dataLayer', 'dataLayer');
+    const envParamsString = envParams ? `&${envParams}`: '';
 
     assert(`[ember-metrics] You must pass a valid \`id\` to the ${this.toString()} adapter`, id);
 
     set(this, 'dataLayer', dataLayer);
 
     if (canUseDOM) {
-      /* jshint ignore:start */
-      (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
-      new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
-      j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
-      '//www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
-      })(window,document,'script',get(this, 'dataLayer'),id);
-      /* jshint ignore:end */
+      (function(w, d, s, l, i) {
```

Comment:
_JSHint_ is a tool used on this project to ensure that the code respects some defined
style guide.
Here, the GTM snippet didn't respect that style guide. So we told _JSHint_ 
to skip this part of the code during its validation process.
This linter is also configured to warn us if we declare some variables without using them.

The issue here is that if we declare a variable outside the `/* jshint ignore */` block
but use it inside, we will have a warning since the linter was explicitly told to ignore this part of the code.
The idea was then to clean the GTM snippet in order to get rid of the `/* jshint ignore */` so that we don't have any _unused variable_ warnings if we need to use a new variable in this block.